src/cmds/scripts/pbs_topologyinfo.py

```diff
@@ -123,36 +96,39 @@ def socketXMLstart(name, attrs):
         except StandardError, e:
             print 'max/map failed: %s' % e
             return
+
+        try:
+            import xml.parsers.expat
+            from xml.parsers.expat import ExpatError
+            ExpatParser = True
+        except ImportError:
+            ExpatParser = False
+
         for name in files:
             pathname = os.sep.join((dirs, name))
-            socketXMLstart.nsockets = 0
-            socketXMLstart.ngpus = 0
-            socketXMLstart.ncoproc = 0
-            socketXMLstart.hwloclatest = 0
-            socketXMLstart.CrayVersion = "0.0"
+            self.reset()
             try:
                 with open(pathname, "r") as f:
+
                     if platform.system() == "Windows":
-                        (socketXMLstart.nsockets, socketXMLstart.ngpus,
-                            socketXMLstart.ncoproc) = reportsockets_win(f)
-                    else:
+                        self.reportsockets_win(f)
+                    elif ExpatParser:
                         try:
-                            socketXMLstart.isCray = False
                             p = xml.parsers.expat.ParserCreate()
                             p.StartElementHandler = socketXMLstart
                             p.ParseFile(f)
                         except ExpatError as e:
                             print "%s:  parsing error at line %d, column %d" \
                                 % (name, e.lineno, e.offset)
+                    else:
+                        self.countsockets(f)
 
                     if options.sockets:
-                        print "%-*s%d" % (maxwidth + 1, name,
-                                          socketXMLstart.nsockets)
+                        print "%-*s%d" % (maxwidth + 1, name, self.nsockets)
                     else:
-                        total = socketXMLstart.nsockets + \
-                            socketXMLstart.ngpus + socketXMLstart.ncoproc
-                        print "%-*s%d" % (maxwidth + 1, name,
-                                          int(math.ceil(total / 4.0)))
+                        self.nnodes += int(math.ceil(self.ndevices / 4.0))
```

Comment:
Can we have function to calculate the above so that it will be useful when formula of calculating node based licenses change in future.